### DISCUSSION TITLE
Configure option for enabling OpenMP

### DISCUSSION THREAD

**Discussion by gcivario** [intel/llvm#6921] [Ideas]
I'm using the compiler for testing the cross-platform compatibility of SYCL codes between various types of accelerators.
Usually, GPU-accelerated codes are very performance oriented, which is why most of them are also OpenMP-parallel.
Therefore when I first tried to compile a code, I've very highly disappointed to discover that the compiler I had just installed using the recommended method wasn't supporting OpenMP. It was preventing me for using it for any sort of real HPC project and I almost gave up on it.

Fortunately, searching a bit deeper, I found [this discussion](https://github.com/intel/llvm/discussions/3759) which permitted me to have both the OpenMP and the SYCL support in the compiler. This is great and it opened up all kinds of possibilities to me. 

Now, since it is so easy to get OpenMP support, why not proposing it as a command line option to the [configure.py](https://github.com/intel/llvm/blob/sycl/buildbot/configure.py) file?
Or at the very least, could the process for enabling OpenMP be documented somewhere?

**Comment by al42and** [comment]
You can use the `--llvm-external-projects openmp` flag when calling `configure.py`. Probably not as easy as using a dedicated option, but it at least does not require modifying any files.

**Comment by zjin-lcf** [comment]
The default OpenMP build option is OpenMP target offloading.
What is the command-line option to disable target offloading when using the script ‘configure.py’ ?   Thanks.

**Reply by al42and** [reply]
If you want to disable offloading support (agree, it sometimes causes troubles during the build), you can add the following flags to `configure.py`: `--cmake-opt=-DLIBOMPTARGET_BUILD_AMDGPU_PLUGIN=FALSE --cmake-opt=-DLIBOMPTARGET_BUILD_CUDA_PLUGIN=FALSE --cmake-opt=-DLIBOMPTARGET_BUILD_DEVICERTL_BCLIB=FALSE`. It tends to vary slightly between LLVM versions, but something along these lines.

**Reply by zjin-lcf** [reply]
I think users are not clear how to add CPU OpenMP build when building the SYCL compiler with CUDA/HIP support. Thanks.

**Reply by zjin-lcf** [reply]
i.e. allows the SYCL compiler to recognize `-fopenmp`
